src/components/checkbox/index.test.js

```diff
@@ -57,15 +63,23 @@ describe('Checkbox', () => {
   describe('events', () => {
     const event = { target: { checked: true } }
 
-    xit('toggles its checked state when changed', () => {
-      expect(wrapper.state().checked).toBeFalsy()
-      wrapper.find('#test').simulate('change', event)
-      expect(wrapper.state().checked).toBeTruthy()
-    })
-
     it('fires the handleChange prop with an event when changed', () => {
-      wrapper.find('#test').simulate('change', event)
+      wrapperSmartShallow.find('#test').simulate('change', event)
       expect(handleChange).toHaveBeenCalledWith(event)
     })
   })
+
+  describe('events', () => {
+    const event = { target: { checked: true } }
+    jest.spyOn(React, 'useEffect').mockImplementation((checked = true))
```

Comment:
There is a [utility](https://reactjs.org/docs/test-utils.html#act) available to trigger `useEffect` in tests. Much easier and safer than mocking the entire function.